azavea/raster-foundry
# Add API/Models for Uploads

## Overview

This commit adds an API endpoint for uploads and updates the swagger
specification to include a metadata field for optional metadata that
will be used during scene creation

### Checklist

~- [ ] Styleguide updated, if necessary~

### Demo

![post-request](https://cloud.githubusercontent.com/assets/898060/24362289/50035d7e-12da-11e7-88d5-fd882e2acc4d.png)
![get-request](https://cloud.githubusercontent.com/assets/898060/24362299/535a6288-12da-11e7-8703-c6e54bc697e1.png)

### Notes

Because this is blocking some work, I'm deferring adding tests for this specific endpoint and creating an issue for that.

## Testing Instructions

 * Apply migrations and start the API server
 * Make a `POST` request to `/api/uploads/` with the following JSON (organization and datasource may need to be adjusted:
```
{
  "organizationId": "dfac6307-b5ef-43f7-beda-b9f208bb7726",
  "uploadStatus": "UPLOADING",
  "fileType": "GEOTIFF",
  "uploadType": "LOCAL",
  "files": [],
  "datasource": "697a0b91-b7a8-446e-842c-97cda155554d",
  "metadata": {}
}
```
 * Make a `GET` request and confirm that filters work
 * Make a `PUT` request and then a `GET` request to confirm that updates work

Connects #1265
